mondrian/src/main/java/mondrian/xmla/impl/DefaultXmlaRequest.java

```diff
@@ -297,6 +308,30 @@ private void initProperties( Element propertiesRoot ) throws XmlaException {
     this.properties = Collections.unmodifiableMap( localProperties );
   }
 
+  private void initParameters( Element parameterElement ) throws XmlaException {
+    Map<String, String> localParameters = new HashMap<>();
+
+    NodeList nlst = parameterElement.getChildNodes();
+    for ( int i = 0, nlen = nlst.getLength(); i < nlen; i++ ) {
+      Node n = nlst.item( i );
+      if ( n instanceof Element ) {
```

Comment:
I would suggest inverting the if conditions to avoid nesting, but up to you.

Reply:
unfortunately sonar doesn't like inverting (leads to java:S135 and then java:S1121), leaving as is for now